tests/rest/admin/test_room.py

```diff
@@ -687,6 +687,8 @@ def test_delete_same_room_twice(self):
         self.assertEqual(200, status_channel.code, msg=status_channel.json_body)
         self.assertEqual("complete", status_channel.json_body["status"])
 
+    test_delete_same_room_twice.skip = "Disabled by default because it's flaky"
```

Comment:
the reason this is flaky is that our database caching layer doesn't necessarily complete its callers in order (it's backed by a `set`) - so when we have multiple requests that each do a `get_user_by_access_token` query, there is no guarantee that one completes before the other.

I think there are two things you could do here:
 * arrange for the purge operation to take some time, then do `first_channel.await_result()` *before* making the second request. This would at least check that the second request cannot start once the first is in progress.
 * Or, allow *either* `first_channel` *or* `second_channel` (but not both) to complete with a 200.

Reply:
alternatively, wait for https://github.com/matrix-org/synapse/pull/11229 to land, which I think will fix this.